### PR TITLE
#19768: Implemented test to check updating track analysis

### PR DESCRIPTION
**Pull request by Corwin-Kh**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/track/helpers/GPXDatabase.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public class GPXDatabase {
 
 	public static final Log LOG = PlatformUtil.getLog(GPXDatabase.class);
 
-	protected static final int DB_VERSION = 23;
+	protected static final int DB_VERSION = 24;
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/track/helpers/GPXDatabase.java:43]: Do we really need to update it for test?

File: OsmAnd/src/net/osmand/plus/track/helpers/GpxDbHelper.java [hunk 2]
```diff
@@ -37,6 +37,11 @@ public class GpxDbHelper implements GpxDbReaderCallback {
 	private final Map<File, GpxDataItemCallback> readingItemsCallbacks = new ConcurrentHashMap<>();
 
 	private GpxReaderTask readerTask;
+	private DBTestCallback dbTestCallback;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/track/helpers/GpxDbHelper.java:40]: static counter will be enough for test, how many reads are done from DB

**Comment by vshcherb** [comment]
Definitely not 4.7 branch

File: OsmAnd/src/net/osmand/plus/track/helpers/GpxDbHelper.java (hunk 2 above)
Comment (Corwin-Kh) [reply OsmAnd/src/net/osmand/plus/track/helpers/GpxDbHelper.java:40]: using callback we ensure we perform this action only in test, checking only specific track and skip analysis update on initial track loading.

**Pull request closed without merging**